server/src/test/java/org/elasticsearch/gateway/ReplicaShardAllocatorIT.java

```diff
@@ -282,6 +285,39 @@ public void testPreferCopyWithHighestMatchingOperations() throws Exception {
         assertThat(internalCluster().nodesInclude(indexName), allOf(hasItem(nodeWithHigherMatching), not(hasItem(nodeWithLowerMatching))));
     }
 
+    /**
+     * Make sure that we do not repeatedly cancel an ongoing recovery for a noop copy on a broken node.
+     */
+    public void testDoNotCancelRecoveryForBrokenNode() throws Exception {
```

Comment:
Just a drive by comment from trying understand this one a little :)

It seems this test isn't really reproducing the issue much better than `testDoNotInfinitelyWaitForMapping` (only fails randomly at a low rate). Could we make this one more deterministic?

Reply:
++. I adjusted this test in https://github.com/elastic/elasticsearch/pull/48265/commits/600c3dee8905b53243d9c12983b365fbbecf151e.